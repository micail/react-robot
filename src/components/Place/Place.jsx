import React from 'react';
import PropTypes from 'prop-types';

import Robot from '../Robot/Robot';

const Place = ({ x, y, position }) => {
  const robotRender = position.x === x && position.y === y ? <Robot f={position.f} /> : null;

  return (
    <div className="place">
      <p>
        {x}
        ,
        {y}
      </p>
      {robotRender}
    </div>
  );
};

export default Place;

const position = {
  x: PropTypes.number,
  y: PropTypes.number,
  f: PropTypes.string,
};

Place.propTypes = {
  y: PropTypes.number,
  x: PropTypes.number,
  position: PropTypes.shape(position),
};

Place.defaultProps = {
  x: null,
  y: null,
  position: {
    x: null,
    y: null,
    f: null,
  },
};
