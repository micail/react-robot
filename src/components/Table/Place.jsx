import React from 'react';
import PropTypes from 'prop-types';

const Place = ({ x, y }) => {
  return (
    <div className="place">
      <p>
        {x}
        ,
        {y}
      </p>
    </div>
  );
};

export default Place;

Place.propTypes = {
  y: PropTypes.number,
  x: PropTypes.number,
};

Place.defaultProps = {
  x: null,
  y: null,
};
