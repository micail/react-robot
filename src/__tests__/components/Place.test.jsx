import React from 'react';
import { shallow, mount } from 'enzyme';

import Place from '../../components/Place/Place';

describe('Place component', () => {
  it('should render place component', () => {
    const component = shallow(<Place />);
    expect(component).toMatchSnapshot();
  });

  it('should render robot on matching coordinates )', () => {
    const x = 5;
    const y = 5;
    const robotsPosition = {
      x: 5,
      y: 5,
      f: 'S',
    };
    const component = mount(<Place x={x} y={y} position={robotsPosition} />);
    const robot = component.find('#robot');
    expect(robot.exists()).toEqual(true);
    expect(robot.hasClass('S')).toEqual(true);
  });

  it('should NOT render robot on matching coordinates )', () => {
    const x = 0;
    const y = 5;
    const robotsPosition = {
      x: 5,
      y: 5,
      f: 'S',
    };
    const component = mount(<Place x={x} y={y} position={robotsPosition} />);
    const robot = component.find('#robot');
    expect(robot.exists()).toEqual(false);
  });
});
