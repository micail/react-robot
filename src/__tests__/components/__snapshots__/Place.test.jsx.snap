// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place component should render place component 1`] = `
<div
  className="place"
>
  <p>
    ,
  </p>
  <Robot
    f={null}
  />
</div>
`;
